### PR TITLE
Modexpw 240-Populate kafka payload with required fileds for Export History

### PR DESCRIPTION
**Pull request by SinghAdes**
## Purpose

https://issues.folio.org/browse/MODEXPW-240
## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/dew/batch/acquisitions/edifact/jobs/ExportHistoryTasklet.java [hunk 1]
```diff
@@ -30,26 +32,40 @@ public class ExportHistoryTasklet implements Tasklet {
...
 
     kafkaService.send(KafkaService.Topic.EXPORT_HISTORY_CREATE, null, exportHistory);
 
     return RepeatStatus.FINISHED;
   }
 
-  ExportHistory buildExportHistory(ChunkContext chunkContext) {
+  ExportHistory buildExportHistory(ChunkContext chunkContext) throws Exception {
+    var jobParameters = chunkContext.getStepContext().getJobParameters();
+    var ediExportConfig = objectMapper.readValue((String)jobParameters.get("edifactOrdersExport"), VendorEdiOrdersExportConfig.class);
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/dew/batch/acquisitions/edifact/jobs/ExportHistoryTasklet.java:50]: As I remember for these values "edifactOrdersExport", "edifactFileName" declared constans, so we can use these constants

File: src/main/java/org/folio/dew/batch/acquisitions/edifact/jobs/ExportHistoryTasklet.java [hunk 2]
```diff
@@ -30,26 +32,40 @@ public class ExportHistoryTasklet implements Tasklet {
...
     var stepExecutionContext = chunkContext.getStepContext().getStepExecution();
     var polineIds = getPoLineIdsFromExecutionContext(stepExecutionContext);
+    var fileName = ExecutionContextUtils.getExecutionVariable(stepExecutionContext, "edifactFileName");
+    var jobStatus = chunkContext.getStepContext().getStepExecution().getJobExecution().getStatus().toString();
 
     return new ExportHistory()
       .id(UUID.randomUUID().toString())
       .exportJobId(jobId)
       .exportDate(new Date())
+      .exportMethod(ediExportConfig.getEdiConfig().toString())
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/dew/batch/acquisitions/edifact/jobs/ExportHistoryTasklet.java:63]: We don't need to put all config, need to put only configName

File: src/main/java/org/folio/dew/batch/acquisitions/edifact/jobs/ExportHistoryTasklet.java [hunk 3]
```diff
@@ -30,26 +32,40 @@ public class ExportHistoryTasklet implements Tasklet {
...
+  ExportHistory buildExportHistory(ChunkContext chunkContext) throws Exception {
+    var jobParameters = chunkContext.getStepContext().getJobParameters();
+    var ediExportConfig = objectMapper.readValue((String)jobParameters.get("edifactOrdersExport"), VendorEdiOrdersExportConfig.class);
+    var vendorId = ediExportConfig.getVendorId().toString();
+    var vendor = organizationsService.getOrganizationById(vendorId);
+    var vendorName = vendor.get("code").asText();
     var stepExecutionContext = chunkContext.getStepContext().getStepExecution();
     var polineIds = getPoLineIdsFromExecutionContext(stepExecutionContext);
+    var fileName = ExecutionContextUtils.getExecutionVariable(stepExecutionContext, "edifactFileName");
+    var jobStatus = chunkContext.getStepContext().getStepExecution().getJobExecution().getStatus().toString();
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/dew/batch/acquisitions/edifact/jobs/ExportHistoryTasklet.java:57]: Did you check if for example job finished with some exception, is job execution status will be failed in this case?

File: src/main/java/org/folio/dew/batch/acquisitions/edifact/jobs/ExportHistoryTasklet.java (hunk 1 above)
Comment (SinghAdes) [reply src/main/java/org/folio/dew/batch/acquisitions/edifact/jobs/ExportHistoryTasklet.java]: fixed

File: src/main/java/org/folio/dew/batch/acquisitions/edifact/jobs/ExportHistoryTasklet.java (hunk 2 above)
Comment (SinghAdes) [reply src/main/java/org/folio/dew/batch/acquisitions/edifact/jobs/ExportHistoryTasklet.java]: fixed

File: src/main/java/org/folio/dew/batch/acquisitions/edifact/jobs/ExportHistoryTasklet.java (hunk 3 above)
Comment (SinghAdes) [reply src/main/java/org/folio/dew/batch/acquisitions/edifact/jobs/ExportHistoryTasklet.java:63]: yes , i have tested with some failed cases and process failed to previous step only. control is not moving to the last step(createExportHistory).

Comment (SerhiiNosko) [reply src/main/java/org/folio/dew/batch/acquisitions/edifact/jobs/ExportHistoryTasklet.java]: so fo now we will send only jobs with Succesful status

**Review by Abdulkhakimov — APPROVED** [review]

**Review by okolawole-ebsco — APPROVED** [review]

**Review by SerhiiNosko — APPROVED** [review]

**Pull request merged**